docs/modules_packages_crates/workspaces.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Workspaces
+---
+
+Workspaces are a feature of nargo that allow you to manage multiple related Noir packages in a single repository. A workspace is essentially a group of related projects that share common build output directories and configurations.
+
+Each Noir project (with it's own Nargo.toml file) can be thought of as a package. Each package is expected to contain exactly one "named circuit", being the "name" defined in Nargo.toml with the program logic defined in `./src/main.nr`.
+
+For a project with the following structure:
+
+```tree
+├── crates
+│   ├── a
+│   │   ├── Nargo.toml
+│   │   └── src
+│   │       └── main.nr
+│   └── b
+│       ├── Nargo.toml
+│       └── src
+│           └── main.nr
+├── Nargo.toml
+└── Prover.toml
+```
+
+You can define a workspace in Nargo.toml like so:
+
+```yaml
+[workspace]
+members = ["crates/a", "crates/b"]
+default-member = "crates/a"
+```
+
+`members` indicates which packages are included in the workspace. This is used when the `--workspace` flag is used with various commands.
```

Comment:
```suggestion
`members` indicates which packages are included in the workspace. As such, all member packages of a workspace will be processed when the `--workspace` flag is used with various commands or if a `default-member` is not specified.
```

members are always members and they are resolved at the beginning of any command. They just aren't processed if a `default-member` is specified.